chapters/20-nodejs.md

```diff
@@ -491,3 +491,61 @@ Objetos que emitem eventos no Node têm um método chamado `on` que é similar
 ao método `addEventListener` no navegador. Você dá um nome de evento e então
 uma função, e isso irá registrar uma função para ser chamada toda vez que um
 dado evento ocorrer.
+
+_Streams_ de leitura possuem os eventos `"data"` e `"end"`. O primeiro é
+acionado sempre que existe alguma informação chegando, e o segundo é chamado
+sempre que a _stream_ chega ao fim. Esse modelo é mais adequado para um
+_streamming_ de dados, que pode ser imediatamente processado, mesmo quando todo
+documento ainda não está disponíevel. Um arquivo pode ser lido como uma _stream_
+de leitura usando a função `fs.createReadStream`.
+
+O seguinte código  cria um servidor que lê o corpo da requisição e o devolve em
+caixa alta para o cliente via _stream_:
+
+```javascript
+var http = require("http");
+http.createServer(function(request, response) {
+  response.writeHead(200, {"Content-Type": "text/plain"});
+  request.on("data", function(chunk) {
+    response.write(chunk.toString().toUpperCase());
+  });
+  request.on("end", function() {
+    response.end();
+  });
+}).listen(8000);
+```
+
+A variável `chunk` enviada para o manipulador de dados será um `Buffer` binário,
+que nós podemos converter para uma _string_ chamando `toString` nele, que vai
+decodificá-lo usando a codificação padrão (UTF-8).
```

Comment:
@gabrielfgularte fiz esta observação pois a pronúncia da frase sobressaiu um pouco melhor. Mas isso não impede caso não for necessário a troca.